src/components/application_manager/src/rpc_handler_impl.cc

```diff
@@ -63,17 +65,46 @@ void RPCHandlerImpl::ProcessMessageFromMobile(
 #endif  // TELEMETRY_MONITOR
   smart_objects::SmartObjectSPtr so_from_mobile =
       std::make_shared<smart_objects::SmartObject>();
-
+  bool remove_unknown_parameters = true;
   DCHECK_OR_RETURN_VOID(so_from_mobile);
   if (!so_from_mobile) {
     LOG4CXX_ERROR(logger_, "Null pointer");
     return;
   }
 
-  if (!ConvertMessageToSO(*message, *so_from_mobile)) {
+  bool rpc_passing = false;
+  if (app_manager_.GetRPCService().CanHandleRPCUsingAppServices(
+          message->function_id(),
+          commands::Command::SOURCE_MOBILE,
+          rpc_passing)) {
+    LOG4CXX_DEBUG(logger_,
+                  "Allowing unknown parameters for request function "
+                      << message->function_id());
+    remove_unknown_parameters = false;
+  }
+
+  if (!ConvertMessageToSO(
```

Comment:
@ShobhitAd For initial `ConvertMessageToSO` implementation unknown parameters is not a problem. 
I mean that it just leaves it inside, isn't it? 
I would propose to extract functionality of removing unknown parameters to a separate function inside AppServices RPC plugin (or any other plugin) if it is required. 
I don't think that is will significantly decrease performance.

Reply:
I'm not certain I understand your comment. The reason for this new parameter is so that it _won't_ remove unknown parameters in some cases, since it is a requirement that we pass unknown parameters through when using app service related functionality. Before this, unknown parameters were automatically removed in the `ConvertMessageToSO` function (https://github.com/smartdevicelink/sdl_core/pull/2821/files#diff-4114af98b73fb870a92d26bdfa94cd66L265). This makes it so it is possible to leave these unknown parameters.